golos.publication/golos.publication.cpp

```diff
@@ -556,17 +556,22 @@ void publication::set_vote(name voter, const structures::mssgid& message_id, int
             send_poolstate_event(item);
         });
         
+        fixp_t sharesfn = get_calc_sharesfn(
+                    pool->rules.mainfunc.code,
+                    FP(new_mssg_rshares.data()),
+                    FP(pool->rules.mainfunc.maxarg)
+                );
+        wdfp_t total_rsharesfn = WP(pool->state.rsharesfn);
```

Comment:
This does not allow solving the problem of calculating payments on Golos.io side. If users vote for another post, this value will not be correct. Why not use `rsharesfn` from the `poolstate` event?